src/frontend/Warnings.ml

```diff
@@ -7,8 +7,8 @@ let pp ?printed_filename ppf (span, message) =
   let loc_str =
     if span = Location_span.empty then ""
     else " in " ^ Location.to_string ?printed_filename span.begin_loc in
-  Fmt.pf ppf "@[<hov>Warning%s: %s@]" loc_str message
+  Fmt.pf ppf "@[<hov 4>Warning%s: %a@]" loc_str Fmt.text message
 
 let pp_warnings ?printed_filename ppf warnings =
   if List.length warnings > 0 then
-    Fmt.(pf ppf "@[<v>%a@]@." (list ~sep:cut (pp ?printed_filename)) warnings)
+    Fmt.(pf ppf "@[<v>%a@]@\n" (list ~sep:cut (pp ?printed_filename)) warnings)
```

Comment:
Why this change? We try to avoid `\n` in format strings because the formatter does not recognize it as a newline.

Reply:
`@\n` and `\n` are not the same, the former works like `@.` but always forces a newline rather than letting the formatter sometimes ignore it. In particular, when ppf is stderr, this guarantees that the formatter always flushes the stream.